app/controllers/application_controller.rb

```diff
@@ -10,6 +10,16 @@ class ApplicationController < ActionController::Base
   before_action :set_locale
 
   rescue_from ActiveRecord::RecordNotFound, with: :render_404
+  before_action :set_csp
+
+  def set_csp
```

Comment:
I would much rather have the logic in the app, so that it can be tested and we aren't relying on Fastly being in use for the headers to be present.